src/core/raps/unlockAndCrosschainSwap.ts

```diff
@@ -51,7 +51,8 @@ export const estimateUnlockAndCrosschainSwap = async (
     isLowerCaseMatch(ETH_ADDRESS_AGGREGATOR, sellTokenAddress) ||
     isNativeAsset(assetToSell.address, chainId);
 
-  if (!isNativeAssetUnwrapping && !nativeAsset) {
+  const shouldNotUnlockAsset = quote.no_approval !== undefined && quote.no_approval;
```

Comment:
I get confused with double negatives `!shouldNotUnlockAsset` is the same as `shouldUnlockAsset`?
no_approval === true means it does not have an approval and need one
no_approval === false means we good to go
no_approval === undefined means what

Reply:
Srry for the double negative. @greg-schrammel I'll try to explain a bit:

`shouldNotUnlockAsset` values:
- `true`: marks that the asset does not need an approval/permit
- `false`: **does NOT mean that the asset requires unlock**. Instead, it means that we **need to check** if the asset requires unlock.

------

- `no_approval === true`: means the asset never needs an approval/permit
- `no_approval === false`: means the asset may need an approval/permit, should be checked
    - `no_approval === undefined`: means the backend did not return this field, so should be treated as it was false.

Reply:
can we change it to `const shouldUnlockAsset = !quote.no_approval;`?